zcash_client_backend/src/data_api.rs

```diff
@@ -441,6 +527,15 @@ pub trait WalletRead {
     /// The type of errors that may be generated when querying a wallet data store.
     type Error;
 
+    /// The type used to track unique account identifiers.
+    type AccountId: Clone + Send + PartialEq + Eq + std::hash::Hash + ConditionallySelectable;
```

Comment:
Requires a `CHANGELOG` entry. Note here that this replaces the existing `AccountId` parameter type in all of the other trait methods.